src/diff_parser/__init__.py

```diff
@@ -72,54 +76,57 @@ def __init__(self, diff: str) -> None:
                     diff = file.readlines()
         else:
             diff = diff.splitlines()
-                
+
         blocks = list()
         block = None
 
         for line in diff:
             if line.startswith("diff --git"):
                 if block:
                     blocks.append(block)
-                block = [line,]
+                block = [line, ]
             else:
                 block.append(line)
-        
+        blocks.append(block)
         filediffs = list()
 
         for block in blocks:
             filediff = DiffBlock()
-            
+
             for line in block:
                 # getting filename and filepath
                 if line.startswith("diff --git"):
                     filenames = line[12:].strip().split(" b/")
-                    if filenames[0][1:] == filenames[1]:
+                    if len(filenames) == 2:
                         filename = filenames[0]
-                        filediff.filepath = filename
-                        filediff.filename = filename.split("/")[-1]
+                        filediff.old_filepath = filename
+                        filediff.old_filename = filename.split("/")[-1]
+
+                        filename = fr"/{filenames[1]}"
```

Comment:
I believe only `f` is enough in fstring no need `r`. There is no regex.

change `fr"/{filenames[1]"` to `f"/{filenames[1]"`

Reply:
by the way "r" is for "raw"
Which I had left because I manually wrote the path to test (And you know we \n and stuff exists)